src/hex_map/iter_radial.h

```diff
@@ -22,13 +22,11 @@ struct HexMapIterRadial : public HexMapIter {
             axial_iter(iter), radius(radius), exclude_center(exclude_center){};
     HexMapIterRadial(const HexMapIterRadial &) = default;
 
-    friend bool operator==(const HexMapIterRadial &a,
-            const HexMapIterRadial &b) {
-        return a.axial_iter.cell == b.axial_iter.cell;
+    bool operator==(const HexMapIterRadial &other) {
+        return this->axial_iter.cell == other.axial_iter.cell;
```

Comment:
From your notes in #3:
> src/hex_map/iter_radial.h: In function 'bool operator==(const HexMapIterRadial&, const HexMapIterR
> src/hex_map/iter_radial.h:27:29: error: 'HexMapCellId HexMapIterAxial::cell' is private within this context
>   27 |         return a.axial_iter.cell == b.axial_iter.cell;
>      |   

What I don't understand here is why the compiler thinks the fields are private when the operator function is marked as friend.  Maybe I need to move the operator outside of the scope of the struct?  But also, why is this only a problem here, but not with the other iterators that I believe follow a similar pattern.

I'll dig into this.

Reply:
Ok, this is another c++20 feature that isn't supported by the compiler.